### PR TITLE
Chore: Cleanup namespace and ID resolution

### LINKED ISSUE #72791: Improve error handling when resolving namespace and ID
We have a [NamespaceID](https://github.com/grafana/grafana/blob/main/pkg/services/authn/authn.go#L236) method which resolve the predefined namespace and ID from the request. 

The resolution can result to either an `error` or parsed ID. When the ID can't be parsed, we default to return `-1`. However when it errors (e.g. because the format of received string is unexpected, or we can't parse to ID) we do not handle error case properly.

This makes it difficult to debug potential problems too.

### PR DESCRIPTION
**Pull request by vtorosyan**
**What is this feature?**

Removes redundant `NamespacedID` and uses existing `GetNamespacedID` method to resolve namespace and ID from the signed in identity.

Note that this was also an action item from the incident we had a while ago.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/72791

**Special notes for your reviewer:**

- As we pushed up parsing of the ID to `int` to the downstream, in many places the parsed ID is not checked to not be zero or negative integer. I have intentionally did not change to have exact same behaviour as before, however I think this is something that we better change, as I can't see any legit scenario when the ID can be zero and we need to act on it.

Changes in the enterprise codebase: https://github.com/grafana/grafana-enterprise/pull/6112

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/authn/authnimpl/service.go [hunk 1]
```diff
@@ -286,10 +286,14 @@ func (s *Service) Login(ctx context.Context, client string, r *authn.Request) (i
 		return nil, err
 	}
 
-	namespace, id := identity.NamespacedID()
+	namespace, id := identity.GetNamespacedID()
+	intId, err := strconv.ParseInt(id, 10, 64)
+	if err != nil || intId <= 0 {
+		return nil, authn.ErrInvalidIdentityID.Errorf("expected correct identity ID but got: %s", id)
+	}
```
Comment (kalleep) [inline pkg/services/authn/authnimpl/service.go:293]: We should replace these with https://github.com/grafana/grafana/blob/main/pkg/services/auth/identity/requester.go#L73

Comment (vtorosyan) [reply pkg/services/authn/authnimpl/service.go:293]: oh nice, I totally missed this. I see there is also `UserIdentifier` which can be used in most of the places as well.

Comment (vtorosyan) [reply pkg/services/authn/authnimpl/service.go:293]: Interesting choice by the way to put the int resolution in the Identity interface - as I was thinking to do something like this but I thought it is going to pollute the Identity interface, as parsing a possible string identity to int is not it's concern and should be done somewhere downstream. 

Curious what do you think, perhaps some helper abstraction outside of identity interface would be an alternative or something like that (not gonna change in this PR anyway)

Comment (kalleep) [reply pkg/services/authn/authnimpl/service.go:293]: Yeah so this is not on the interface but just a helper function with common patterns

**Comment by vtorosyan** [comment]
Ready to be reviewed again - thanks for the feedback!

**Review by kalleep — APPROVED** [review]
LGTM

**Review by Jguer — APPROVED** [review]

**Pull request merged**